### PR TITLE
 ft-#167370397 allow admin and moderator to block or unblock reported comments

### PR DESCRIPTION
**Pull request by Ramadhan0**
## What does this PR do?
this PR let the admin/moderator be able to block or unblock reported comment
[usecase](https://docs.google.com/document/d/1MWq5TFvgFjr2UzTv-GmYEYp756pvgRkMhchmyo_w3nM/edit?usp=sharing)
#### Description of Task to be completed?
Have the following endpoint working
`PATCH` : `/api/admin/comment/:id/block`
`PATCH` : `/api/admin/comment/:id/unblock`

#### How should this be manually tested?

- Clone the repo to the local machine.
- Run  :
`npm install` 
`npm setup:dev`
`npm start.`

- Send a PATCH request to :
`/api/admin/comment/:id/block`
 `/api/admin/comment/:id/unblock`

#### What are the relevant pivotal tracker stories?
[16370397](https://www.pivotaltracker.com/story/show/167370397)

### CODE REVIEW

**Comment by alainburindi** [comment]
rename it properly

File: src/api/controllers/reportedCommentAdminController.js
Comment (alainburindi) [inline src/api/controllers/reportedCommentAdminController.js]: is this the name of the class?

File: src/api/controllers/reportedCommentAdminController.js
Comment (alainburindi) [inline src/api/controllers/reportedCommentAdminController.js]: is this data needed? what do you think

File: src/api/controllers/reportedCommentAdminController.js
Comment (alainburindi) [inline src/api/controllers/reportedCommentAdminController.js]: is it @memberof blockComment?

File: src/api/controllers/reportedCommentAdminController.js
Comment (alainburindi) [inline src/api/controllers/reportedCommentAdminController.js]: is it a string?

File: src/api/routes/adminRouter.js
Comment (alainburindi) [inline src/api/routes/adminRouter.js]: what do you think about `comments/:id/block`

File: src/middlewares/checkIfCommentExist.js
Comment (alainburindi) [inline src/middlewares/checkIfCommentExist.js]: @memberof checkIfCommentExist

**Review by alainburindi — CHANGES_REQUESTED** [review]
as I see, you blocked the comment but they can be accessed using the GET comments route.
What do you think if the admin wants to see the blocked comments?

**Comment by nkpremices** [comment]
@Ramadhan0 I think you need to add more steps in the description of the PR `how should this be manually tested`. For e.g what to do to see if a comment can be blocked or not

File: src/api/controllers/reportedCommentAdminController.js
Comment (premlogic) [inline src/api/controllers/reportedCommentAdminController.js]: @Ramadhan0 I suggest the name of this function to just be `commentReportController`

File: src/api/controllers/reportedCommentAdminController.js
Comment (premlogic) [inline src/api/controllers/reportedCommentAdminController.js]: `A class containing all controllers to block and unblock comments`

File: src/api/controllers/reportedCommentAdminController.js
Comment (premlogic) [inline src/api/controllers/reportedCommentAdminController.js]: @Ramadhan0 I really agree that the logic here is good and everything is working perfectly. 
I just have sort of an idea to improve what's happening here. 

I can see that the two queries you are sending to the database are exactly going to the same key.  What if we just use one then we find a way of sending defferent values to the controller according to what route was used ?

File: src/middlewares/checkIfCommentExist.js
Comment (premlogic) [inline src/middlewares/checkIfCommentExist.js]: `A callback to allow the node process to continue`

File: src/middlewares/checkIfCommentExist.js
Comment (premlogic) [inline src/middlewares/checkIfCommentExist.js:21]: I have a middleware doing almost the same thing. I think there is a way we can merge them into one

File: src/middlewares/isCommentBlocked.js
Comment (premlogic) [inline src/middlewares/isCommentBlocked.js:1]: this also could be used with the same middleware. 
I really have an Idea

**Review by nkpremices — CHANGES_REQUESTED** [review]
I left some comments @Ramadhan0

File: src/api/controllers/reportedCommentAdminController.js
Comment (MemunaHaruna) [reply src/api/controllers/reportedCommentAdminController.js]: @Ramadhan0 you haven't addressed this feedback. you can place all admin related controllers in a folder called admin like:

`src/api/controllers/admin/commentReportController`

File: src/docs/swagger.json
Comment (MemunaHaruna) [inline src/docs/swagger.json]: @Ramadhan0 edit this sentence.

File: src/docs/swagger.json
Comment (MemunaHaruna) [inline src/docs/swagger.json]: @Ramadhan0 modify what?

File: src/docs/swagger.json
Comment (MemunaHaruna) [inline src/docs/swagger.json]: same here. this sentence needs editing

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: src/middlewares/checkIfCommentExist.js
Comment (Ramadhan0) [reply src/middlewares/checkIfCommentExist.js:21]: @truestbyheart has a function to feed all middlewares,
I think after @MemunaHaruna merge this PR, this middleware will get refactored.

File: src/middlewares/isCommentBlocked.js
Comment (Ramadhan0) [reply src/middlewares/isCommentBlocked.js:1]: @nkpremices, we have a simple way to refactor this after @MemunaHaruna merge

**Review by MemunaHaruna — APPROVED** [review]

**Pull request merged**
